lesson25/main.js

```diff
@@ -19,14 +32,84 @@ submitBtn.addEventListener(
   false
 );
 
+const checkInputVal = (targetForm) => {
+  const targetVal = targetForm.value;
+  let result;
+
+  switch (targetForm.id) {
+    case "userName":
+      const minLength = 1;
+      const maxLength = 15;
+      result = targetVal.length >= minLength && targetVal.length <= maxLength;
+      break;
+    case "email":
+      const emailValid = /^\w[\w_.-]*@.+\..+/;
+      result = emailValid.test(targetVal);
+      break;
+    case "pass":
+      const passValid = /(?=.*[a-z])(?=.*[A-Z])(?=.*\d)\w{8,}/;
+      result = passValid.test(targetVal);
+      break;
+    default:
+      console.error(`は見つかりませんでした`);
+  }
+
+  result ? (flags[targetForm.id] = true) : (flags[targetForm.id] = false);
+  return result;
+};
+
+const createErrorMessage = (targetFormId) => {
+  const errorTxt = createElementWithClassName("p", "error-txt");
+  errorTxt.id = "js-errorTxt";
+
+  switch (targetFormId) {
+    case "userName":
+      errorTxt.textContent = "ユーザー名は15文字以下にしてください。";
+      break;
+    case "email":
+      errorTxt.textContent = "メールアドレスの形式になっていません。";
+      break;
+    case "pass":
+      errorTxt.textContent =
+        "8文字以上の大小の英数字を交ぜたものにしてください。";
+      break;
+    default:
+      console.error(`は見つかりませんでした`);
+  }
+
+  return errorTxt;
+};
+
+const renderErrorMessage = (targetFormId) => {
+  const errorTxt = createErrorMessage(targetFormId);
+  const targetInput = document.getElementById(targetFormId);
+  targetInput.after(errorTxt);
+};
```

Comment:
I put it in the createErrorMessage.
```suggestion
```

Reply:
I've deleted this part! 👍 
[2761d46](https://github.com/yuka830/js-lessons/pull/30/commits/2761d462d31116b55a9625f5f70e7db8a411f962)